server/controllers/profile.js

```diff
@@ -41,3 +41,49 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
     },
   });
 });
+
+// @route POST /profile/availability/edit
+// @desc edit user profile
+// @access Private
+exports.editAvailability = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.body.user.id);
+  if (!user) {
+    res.status(404);
+    throw new Error("User doesn't exist");
+  }
+  const profile = await Profile.findById(user.profile);
+  const availableTime = profile.availableTime; //creates variable representing user's availability
+  const newDay = req.body.newDay;
+
+  if (availableTime.startTime > availableTime.endTime) {
+    res.status(401);
+    throw new Error("Start time must occur before end time");
+  }
+
+  if (
+    //checks if newDay is already in availableTime
+    availableTime.some(
+      (day) =>
+        day.date.getDate() === new Date(newDay.date).getDate() &&
+        day.date.getMonth() === new Date(newDay.date).getMonth() &&
+        day.date.getFullYear() === new Date(newDay.date).getFullYear()
+    )
+  ) {
+    //if match, find match and set to newDay (updates match)
+    availableTime.forEach((day) => {
+      if (day.date === newDay.date) day = newDay;
```

Comment:
@jb162  `day.date` and  `newDay.date` are not of the same type, you may want to parse `newDay.date` into a date object to perform a check here :)

Eg: 
```
    if (day.date.getDate() === new Date(newDay.date).getDate()
        .....
       )
```

Reply:
I just added in this change, however, I am still facing the issue of only being able to submit the first update and but no subsequent updates.

Reply:
@jb162 My guess was right, the reason subsequent updates are not applying is because we are using `forEach` - which is read-only.  You can instead use an Array.map() 😄

<img width="844" alt="Screen Shot 2022-01-12 at 7 22 46 PM" src="https://user-images.githubusercontent.com/75184965/149153546-dbba63bc-39d0-4364-bfd5-0fc3294f2a9c.png">